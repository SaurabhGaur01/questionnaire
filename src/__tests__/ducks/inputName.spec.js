import {
    setName,
    clearName,
    inputNameReducer,
} from '../../ducks/userInput/inputName';

describe('inputName Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setName should match the snapshot' , () => {
            expect(setName(mockObject)).toMatchSnapshot();
        });
        it('clearName should match the snapshot' , () => {
            expect(clearName()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(inputNameReducer()).toEqual("");
        });
        it('should return the passed state if called with no action' , () => {
            expect(inputNameReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(inputNameReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(inputNameReducer('', setName(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearName' , () => {
            expect(inputNameReducer('', clearName())).toEqual("");
        });
    });
});