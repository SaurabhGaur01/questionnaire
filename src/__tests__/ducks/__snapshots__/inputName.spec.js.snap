// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputName Duck Tests Action Generator clearName should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_NAME",
}
`;

exports[`inputName Duck Tests Action Generator setName should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_NAME",
  "value": "mockValue",
}
`;
